numba/stencil.py

```diff
@@ -542,6 +542,14 @@ def _stencil_wrapper(self, result, sigret, return_type, typemap, calltypes, *arg
                 out_init ="{} = np.zeros({}, dtype=np.{})\n".format(
                             out_name, shape_name, return_type_name)
             func_text += "    " + out_init
+        else: # result is present, if cval is set then use it
+            if "cval" in self.options:
+                cval = self.options["cval"]
+                if return_type.dtype != typing.typeof.typeof(cval):
```

Comment:
Does the `# TODO: Loosen this restriction to adhere to casting rules.` applies here as well?

Reply:
yes